### ISSUE TITLE
Error when buying from details page

### ISSUE THREAD

**Issue by damienen** [Itheum/data-dex#642]
> Wallet: Xportal
> 
> Issue while purchasing a dataset from recent Data NFTs.
> 
> Upon Acquiring a data NFT the below error is seen.
> 
> Steps to reproduce:
> 
> Browse to dashboard Go to recent Data NFTs Purchase the NFT at https://deploy-preview-583--frosty-hermann-85ca4f.netlify.app/datanfts/marketplace/DATANFTFT4-3ba099-2a/offer-1923 Recent Data NFTs ![image](https://user-images.githubusercontent.com/121969933/238425896-4faaeeaf-5ed7-4821-8698-cbd8d8e87074.png)
> 
> Comment Added: This seems to happen in the detail page if you are clearing out the final NFTs. For example
> 
> If there is 1 NFT left, then purchasing the NFT will result in the error. If there are 2 NFTs left, then purchasing 1 will not throw the error as there will be a balance remaining.

**Comment by icegriffinguru** [comment]
@damienen 
This offer is already bought and closed state
what should we do with such case on NFT detail page?

**Comment by damienen** [comment]
Act the same way the details page acts when being opened from wallet (like
no offer is provided)

**Comment by icegriffinguru** [comment]
![image](https://github.com/Itheum/data-dex/assets/97042732/6b37a485-e0dc-4a0d-8c11-d80c9e024a9e)

It already does not show "Purchase" button or any other details.
The issue is that an error message pops up because viewOffer method panics.
Showing "Offer is closed" or something similar is better.
And how to know if an offer is closed instead getting an error from viewOffer panic?

**Comment by damienen** [comment]
I'm not 100% which would be the best course of action. One idea I am having is:
1. Have a state variable called hasJustBoughtAllSupply that starts as false
2. When sending a transaction mark hasJustBroughtAllSupply as true
3. Do not try to fetch the offer details (or expect to get that error and treat it in order to not receive the toast) is hasJustBroughtAllSupply is true

**Comment by icegriffinguru** [comment]
> I'm not 100% which would be the best course of action. One idea I am having is:
> 
> 1. Have a state variable called hasJustBoughtAllSupply that starts as false
> 2. When sending a transaction mark hasJustBroughtAllSupply as true
> 3. Do not try to fetch the offer details (or expect to get that error and treat it in order to not receive the toast) is hasJustBroughtAllSupply is true

What if a user directly enters the link like me?
It still shows an error popup because offer is closed.

**Comment by icegriffinguru** [comment]
@damienen

**Comment by icegriffinguru** [comment]
```
Issue while purchasing a dataset from recent Data NFTs.
Upon Acquiring a data NFT the below error is seen.
```
The original issue was that a user saw popup error.
We need to think of a better way to show users thatn the offer is sold out.

**Comment by icegriffinguru** [comment]
@damienen , @fuzzyduke . Fixed.

**Comment by fuzzyduke** [comment]
Verified, fixed on #649
